test/integration/users_signup_test.rb

```diff
@@ -0,0 +1,26 @@
+require 'test_helper'
+class UsersSignupTest < ActionDispatch::IntegrationTest
+  test "validate that signups that dont meet form criteria fail" do
+    get signup_path
+    assert_no_difference 'User.count' do
+      post users_path, params: { user: { name:  "",
+                                         email: "user@invalid",
+                                         password: "foo",
+                                         password_confirmation: "bar" } }
+    end
+    assert_template 'users/new'
+  end
+
+  test "test to confirm with valid signup information" do
+    get signup_path
+    assert_difference 'User.count', 1 do
+      post users_path, params: { user: { name: "Example User",
+                                         email: "user@example.com",
+                                         password:"password",
+                                         confirm_password:"password" }}
```

Comment:
```suggestion
      post users_path, params: params
```
extract params into a well formatted variable & use that, ex:
```rb
params = {
  user: {
    ...
  }
}
```